app/controllers/admin/merchants_controller.rb

```diff
@@ -7,4 +7,38 @@ def index
   def show
     @merchant = Merchant.find(params[:id])
   end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    merchant = Merchant.find(params[:id])
+
+    if params[:status].present?
+      merchant.update(admin_merchant_params)
+      merchant.save
+      redirect_to admin_merchants_path
+    else
+      merchant.update(admin_merchant_params)
+      if merchant.save
+        redirect_to admin_merchant_path(merchant.id)
+        flash[:success] = "The information has been successfully updated for #{merchant.name}"
+      end
+    end
+  end
+  
+  def new
+  end
+
+  def create
+    merchant = Merchant.new(admin_merchant_params)
+    merchant.save
+    redirect_to admin_merchants_path
+  end
```

Comment:
Yes! for the sad_path I believe we could add something similar to this in the controller:
```
if merchant.save
      redirect_to admin_merchants_path
    else
      redirect_to edit_admin_merchant_path
      flash[:alert] = "Error: #{error_message(merchant.errors)}"
    end
```

Reply:
Perfect! Yeah, we can get this sad path set up for both that create and edit section for the admin merchants!

Reply:
Also then in the views/layouts/application.html.erb

`<% flash.each do |type, msg| %>
    <div>
      <%= msg %>
    </div>`